### PR TITLE
Adding a contributing and maintenance guide.

### PR DESCRIPTION
**Pull request by Mariatta**
This is just a starting point. I'm sure we can keep on improving it.
Personally for me, I just needed to have the "how to release" written up
since I keep forgetting how to do it.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+- Ensure that the ``__version__`` number in ``gidgethub/__init__.py`` is up to date.
+
+- Ensure the tests have passed and that coverage is at 100%.
+
+Preparing for a release:
+
+- Create a new branch for the release.
+
+- Push the "release" branch to remote.
+
```
Comment (brettcannon) [inline CONTRIBUTING.rst:86]: I actually never do this. 😄  What I do is I start a branch locally with what I _think_ the final state will be, run the tests, do the build, upload, commit my changes, and then create a GH release (so it creates the git tag).

**Review by brettcannon — APPROVED** [review]
LGTM, although I follow a different approach when doing the final changes for the release.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (Mariatta) [reply CONTRIBUTING.rst:86]: Do you use GitHub CLI to release then? I have only been using the web UI, and I needed to either push the branch or the tag in order to do it from the web.

Comment (brettcannon) [reply CONTRIBUTING.rst:86]: Nope, web UI. The Releases page will create a new tag from the point that any branch is currently at if the name doesn't already exist.

**Pull request merged**